contracts/Core/BlockManager.sol

```diff
@@ -117,8 +118,34 @@ contract BlockManager is Initializable, ACL, BlockStorage {
         disputes[epoch][msg.sender] = Structs.Dispute(0, 0, 0, 0);
     }
 
-    function getBlock(uint256 epoch) external view returns (Structs.Block memory _block) {
-        return (blocks[epoch]);
+    function claimBlockReward() external initialized checkState(parameters.confirm()) {
+        uint256 epoch = parameters.getEpoch();
+        uint256 stakerId = stakeManager.getStakerId(msg.sender);
+        require(stakerId > 0, "Structs.Staker does not exist");
+        require(blocks[epoch].proposerId == 0, "Block already confirmed");
+
+        for (uint8 i = 0; i < proposedBlocks[epoch].length; i++) {
+            if (!proposedBlocks[epoch][i].valid) {
+                continue;
+            }
+            require(proposedBlocks[epoch][i].proposerId == stakerId, "Block can be confirmed by proposer of the block");
+            _confirmBlock(epoch, i);
+            rewardManager.giveBlockReward(stakerId, epoch);
+            return;
+        }
+    }
+
+    function confirmBlock(uint256 stakerId) external initialized onlyRole(parameters.getBlockConfirmerHash()) {
+        uint256 epoch = parameters.getEpoch();
+
+        for (uint8 i = 0; i < proposedBlocks[epoch - 1].length; i++) {
+            if (!proposedBlocks[epoch - 1][i].valid) {
```

Comment:
This for loop is common in both claimBlockReward and confirmBlock with only one diff : epoch
can we move it out with epoch as param

Reply:
Anyways i dont think so this for loop will be required when hrishi's PR get merged so this should not be an issue right now

Reply:
okay cool